stl/inc/spanstream

```diff
@@ -0,0 +1,396 @@
+    using char_type   = _Elem;
+    using int_type    = typename _Traits::int_type;
+    using pos_type    = typename _Traits::pos_type;
+    using off_type    = typename _Traits::off_type;
+    using traits_type = _Traits;
+
+    // [spanbuf.ctor], constructors
+    basic_spanbuf() = default;
+
+    explicit basic_spanbuf(ios_base::openmode _Which) : _Mysb(), _Mode(_Which) {}
+
+    explicit basic_spanbuf(_STD span<_Elem> _Span, ios_base::openmode _Which = ios_base::in | ios_base::out)
+        : _Mysb(), _Mode(_Which), _Buf(_Span) {
+        if (_Mode & ios_base::out) {
+            if (_Mode & ios_base::ate) {
+                _Mysb::setp(_Buf.data(), _Buf.data() + _Buf.size(), _Buf.data() + _Buf.size());
+            } else {
+                _Mysb::setp(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+            }
+        }
+
+        if (_Mode & ios_base::in) {
+            _Mysb::setg(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+        }
+    }
+
+    basic_spanbuf(const basic_spanbuf&) = delete;
+    basic_spanbuf(basic_spanbuf&& _Right)
+        : _Mysb(_STD move(_Right)), _Mode(_STD move(_Right._Mode)), _Buf(_STD move(_Right._Buf)) {
+        _Right.setp(nullptr, nullptr, nullptr);
+        _Right.setg(nullptr, nullptr, nullptr);
+    }
+
+    // [spanbuf.assign], assignment and swap
+    basic_spanbuf& operator=(const basic_spanbuf&) = delete;
+
+    basic_spanbuf& operator=(basic_spanbuf&& _Right) {
+        basic_spanbuf tmp{move(_Right)};
+        this->swap(tmp);
+        return *this;
+    }
+
+    void swap(basic_spanbuf& _Right) {
+        _Mysb::swap(_Right);
+        _STD swap(_Mode, _Right._Mode);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    // [spanbuf.members], member functions
+    _STD span<_Elem> span() const noexcept {
+        if (_Mode & ios_base::out) {
+            return _STD span<_Elem>(_Mysb::pbase(), _Mysb::pptr());
+        }
+
+        return _Buf;
+    }
+
+    void span(_STD span<_Elem> _Span) noexcept {
+        _Buf = _Span;
+        if (_Mode & ios_base::out) {
+            if (_Mode & ios_base::ate) {
+                _Mysb::setp(_Buf.data(), _Buf.data() + _Buf.size(), _Buf.data() + _Buf.size());
+            } else {
+                _Mysb::setp(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+            }
+        }
+
+        if (_Mode & ios_base::in) {
+            _Mysb::setg(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+        }
+    }
+
+protected:
+    // [spanbuf.virtuals], overridden virtual functions
+    pos_type __CLR_OR_THIS_CALL seekoff(
+        off_type _Off, ios_base::seekdir _Way, ios_base::openmode _Which = ios_base::in | ios_base::out) override {
+        const bool _Sequence_in  = _Which & ios_base::in;
+        const bool _Sequence_out = _Which & ios_base::out;
+        switch (_Way) {
+        case ios_base::beg:
+            {
+                // [spanbuf.virtuals]/4.1 baseoff = 0

```

Comment:
All of the citations to specific paragraphs need to be qualified with N4892, in case the paragraphs are renumbered in the future. Occurs repeatedly below (I just searched for `]/`). Mentions of stable names alone are fine when unqualified.